### PR TITLE
1229-faultToleranceDoesntWorkwithMetrics

### LINKED ISSUE #1229: MicroProfile: Metrics and FT don't work together
@davidcurrie hit this exception when trying to add @Metered to a class which also has a @Fallback method.

```
Stack Dump = org.jboss.weld.exceptions.DefinitionException: internal.error.CWMFT4999E
    at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessAnnotatedType(ContainerLifecycleEvents.java:182)
    at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessAnnotatedType(ContainerLifecycleEvents.java:166)
    at org.jboss.weld.bootstrap.BeanDeployer.processAnnotatedTypes(BeanDeployer.java:156)
    at org.jboss.weld.bootstrap.BeanDeployment.createTypes(BeanDeployment.java:245)
    at org.jboss.weld.bootstrap.WeldStartup.startInitialization(WeldStartup.java:407)
    at org.jboss.weld.bootstrap.WeldBootstrap.startInitialization(WeldBootstrap.java:76)
    at com.ibm.ws.cdi.impl.CDIContainerImpl$2.run(CDIContainerImpl.java:142)
    at com.ibm.ws.cdi.impl.CDIContainerImpl$2.run(CDIContainerImpl.java:139)
    at java.security.AccessController.doPrivileged(AccessController.java:638)
    at com.ibm.ws.cdi.impl.CDIContainerImpl.applicationStarting(CDIContainerImpl.java:139)
    at com.ibm.ws.cdi.liberty.CDIRuntimeImpl.applicationStarting(CDIRuntimeImpl.java:382)
    at com.ibm.ws.container.service.state.internal.ApplicationStateManager.fireStarting(ApplicationStateManager.java:28)
    at com.ibm.ws.container.service.state.internal.StateChangeServiceImpl.fireApplicationStarting(StateChangeServiceImpl.java:50)
    at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.preDeployApp(DeployedAppInfoBase.java:380)
    at com.ibm.ws.app.manager.module.internal.DeployedAppInfoBase.deployApp(DeployedAppInfoBase.java:409)
    at com.ibm.ws.app.manager.war.internal.WARApplicationHandlerImpl.install(WARApplicationHandlerImpl.java:65)
    at com.ibm.ws.app.manager.internal.statemachine.StartAction.execute(StartAction.java:140)
    at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.enterState(ApplicationStateMachineImpl.java:1258)
    at com.ibm.ws.app.manager.internal.statemachine.ApplicationStateMachineImpl.run(ApplicationStateMachineImpl.java:873)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1160)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
    at java.lang.Thread.run(Thread.java:811)
Caused by: java.lang.RuntimeException: internal.error.CWMFT4999E
    at com.ibm.ws.microprofile.faulttolerance.cdi.AnnotatedTypeWrapper.<init>(AnnotatedTypeWrapper.java:47)
    at com.ibm.ws.microprofile.faulttolerance.cdi.FaultToleranceCDIExtension.addFaultToleranceAnnotation(FaultToleranceCDIExtension.java:170)
    at com.ibm.ws.microprofile.faulttolerance.cdi.FaultToleranceCDIExtension.processAnnotatedType(FaultToleranceCDIExtension.java:119)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:90)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:55)
    at java.lang.reflect.Method.invoke(Method.java:508)
    at org.jboss.weld.injection.StaticMethodInjectionPoint.invoke(StaticMethodInjectionPoint.java:88)
    at org.jboss.weld.injection.MethodInvocationStrategy$SpecialParamPlusBeanManagerStrategy.invoke(MethodInvocationStrategy.java:144)
    at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:299)
    at org.jboss.weld.event.ExtensionObserverMethodImpl.sendEvent(ExtensionObserverMethodImpl.java:124)
    at org.jboss.weld.event.ObserverMethodImpl.sendEvent(ObserverMethodImpl.java:277)
    at org.jboss.weld.event.ObserverMethodImpl.notify(ObserverMethodImpl.java:255)
    at org.jboss.weld.event.ObserverNotifier.notifySyncObservers(ObserverNotifier.java:269)
    at org.jboss.weld.event.ObserverNotifier.notify(ObserverNotifier.java:258)
    at org.jboss.weld.event.ObserverNotifier.fireEvent(ObserverNotifier.java:166)
    at org.jboss.weld.bootstrap.events.ContainerLifecycleEvents.fireProcessAnnotatedType(ContainerLifecycleEvents.java:180)
    ... 21 more
```

### PR DESCRIPTION
**Pull request by tevans78**
#build
fixes #1229

### CODE REVIEW

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/microprofile-config.properties [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# Contributors:
+#     IBM Corporation - initial API and implementation
+###############################################################################
+com.ibm.ws.microprofile.faulttolerance_fat.cdi.beans.FallbackBeanWithoutRetry/connectB/Fallback/value=com.ibm.ws.microprofile.faulttolerance_fat.cdi.beans.MyFallbackHandler2
+com.ibm.ws.microprofile.faulttolerance_fat.cdi.beans.FallbackBeanWithoutRetry/connectC/Fallback/fallbackMethod=connectFallback2
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/microprofile-config.properties]: I don't think this file is needed. None of the referenced beans appear to be in the new test project.

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/permissions.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+
+<permissions xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee http://xmlns.jcp.org/xml/ns/javaee/permissions_7.xsd"
+	version="7">
+        <!-- this permission shouldn't be needed but there is a java bug -->
+        <permission>
```
Comment (Azquelt) [inline dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/permissions.xml]: Is this java bug workaround still needed?

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/microprofile-config.properties (hunk 1 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/microprofile-config.properties]: Fair enough ... will fix under another PR.

File: dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/permissions.xml (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics/test-applications/CDIFaultToleranceMetrics.war/resources/META-INF/permissions.xml]: I don't know but we should check so I will open an issue to do so.

**Comment by tevans78** [comment]
#build

**Comment by tevans78** [comment]
Good PB, merging

**Pull request merged**